### PR TITLE
Update daily min temperatures link for demo dataset

### LINKED ISSUE #3024: Update daily min weather dataset with new link

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #3024

### CODE REVIEW

**Review by angela97lin — APPROVED** [review]
Thank you for doing this, looks good! 🙏 

Can we remove this from `setup.py` as well?

The lines:
```
    data_files=[('evalml/tests/data', ['evalml/tests/data/tips.csv',
                                       'evalml/tests/data/titanic.csv',
                                       'evalml/tests/data/churn.csv',
                                       'evalml/tests/data/fraud_transactions.csv.gz']),
                ('evalml/demos/data', ['evalml/demos/data/daily-min-temperatures.csv'])],
```

**Comment by gsheni** [comment]
@angela97lin done:
https://github.com/alteryx/evalml/pull/3023/commits/80a28053e4532f4e3651a0633702cdf85ce86660

File: setup.py [hunk 1]
```diff
@@ -27,10 +27,5 @@
         'console_scripts': [
           'evalml = evalml.__main__:cli'
         ]
-    },
-    data_files=[('evalml/tests/data', ['evalml/tests/data/tips.csv',
-                                       'evalml/tests/data/titanic.csv',
-                                       'evalml/tests/data/churn.csv',
-                                       'evalml/tests/data/fraud_transactions.csv.gz']),
-                ('evalml/demos/data', ['evalml/demos/data/daily-min-temperatures.csv'])],
```
Comment (freddyaboulton) [inline setup.py:35]: I think we should undo this change and add daily-min-temperatures to the list of `evalml/tests/data`. We need this so the data gets published with the pypi package and conda can run the tests.

Comment (freddyaboulton) [reply setup.py:35]: I think we can remove the copy of daily-min-temperatures in demos/data from version control too.

**Review by freddyaboulton — APPROVED** [review]
Thank you @gsheni !

File: setup.py (hunk 1 above)
Comment (gsheni) [reply setup.py:35]: Fixed
https://github.com/alteryx/evalml/pull/3023/commits/0b7293b5ecb2fa9d0d002006bcaeda29b59ad83d
